AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Features/Itineraries/RemoveRequestCommandHandlerShould.cs

```diff
@@ -0,0 +1,163 @@
+            };
+
+
+            Context.Add(itinerary1);
+            Context.Add(request1);
+            Context.Add(request2);
+            Context.Add(request3);
+            Context.Add(itineraryRequest1);
+            Context.Add(itineraryRequest2);
+            Context.Add(itineraryRequest3);
+            Context.SaveChanges();
+        }
+
+        [Fact]
+        public async Task RemoveItineraryRequestDoesExist()
+        {
+            var mockMediator = new Mock<IMediator>();
+
+            var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
+            var succeded = await handler.Handle(new RemoveRequestCommand
+                { ItineraryId = 1, RequestId = Request1Id });
+            Assert.False(Context.ItineraryRequests.Any(x=>x.RequestId == Request1Id));
+        }
+
+        [Fact]
+        public async Task AbortRemoveItineraryRequestDoesnotExist()
+        {
+            var mockMediator = new Mock<IMediator>(MockBehavior.Strict);
+            var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
+            var succeded = await handler.Handle(new RemoveRequestCommand
+                { ItineraryId = 1, RequestId = new Guid("0da127ec-e4b6-44df-abd3-8a9ffa685826") });
+        }
+
+
+        [Fact]
+        public async Task AbortRemoveCompletedItineraryRequest()
+        {
+            var mockMediator = new Mock<IMediator>();
+            
+            var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
+            var succeded = await handler.Handle(new RemoveRequestCommand
+                { ItineraryId = 1, RequestId = Request2Id });
+            Assert.True(Context.ItineraryRequests.Any(x => x.RequestId == Request1Id));
+        }
+
+        [Fact]
+        public async Task RemoveItineraryRequest_ShouldMoveUpOthers()
+        {
+            var mockMediator = new Mock<IMediator>();
+
+            var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
+            var succeded = await handler.Handle(new RemoveRequestCommand
+                { ItineraryId = 1, RequestId = Request1Id });
+            Assert.Equal(1, Context.ItineraryRequests.First(x => x.RequestId == Request2Id).OrderIndex);
+            Assert.Equal(2, Context.ItineraryRequests.First(x => x.RequestId == Request3Id).OrderIndex);
+        }
+
+        [Fact]
+        public async Task RemoveItineraryRequest_ShouldUnassignRequest()
+        {
+            var mockMediator = new Mock<IMediator>();
+
+            var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
+            var succeded = await handler.Handle(new RemoveRequestCommand
+                { ItineraryId = 1, RequestId = Request1Id });
+            Assert.Equal(RequestStatus.Unassigned, Context.Requests.First(x => x.RequestId == Request1Id).Status);
+        }
+
+
+        // TODO: fix this unit test to be able to check that notification has correct data
```

Comment:
This is testable. Here is the code you would use to verify that `RequestStatusChangedNotification` was published with the correct data
```
var oldStatus = Context.Requests.First(x => x.RequestId == Request1Id).Status;

var mockMediator = new Mock<IMediator>();
var message = new RemoveRequestCommand { ItineraryId = 1, RequestId = Request1Id };var handler = new RemoveRequestCommandHandler(Context, mockMediator.Object);
await handler.Handle(message);

var newStatus = Context.Requests.First(x => x.RequestId == Request1Id).Status;

mockMediator.Verify(x => x.PublishAsync(It.Is<RequestStatusChangedNotification>(y => y.RequestId == message.RequestId && y.OldStatus == oldStatus && y.NewStatus == newStatus)));
```